### PR TITLE
Major revision of group-aws docs

### PR DESCRIPTION
**Pull request by jillr**
With the move to collections we've made a number of changes.
In addition to new paths to issues, PRs, etc we're also implementing
a project board to track work that was historically manually tracked
in this repo (boto refactors, missing ci tests, etc).  Also adds
updates to best practices docs and other housekeeping updates.

### CODE REVIEW

File: group-aws/integration.md [hunk 1]
```diff
@@ -12,167 +12,28 @@ shall be considered a feature request.
...
+* Resource creation again (idempotency) under check mode
+* Resource creation again (idempotency)
+* Resource modification under check mode
 * Resource modification
+* Resource modification again (idempotency) under check mode
+* Resource modification again (idempotency)
+* Resource deletion under check mode
 * Resource deletion
+* Resource deletion (of a non-existent resource) under check mode
+* Resource deletion (of a non-existent resource)
```
Comment (tremble) [inline group-aws/integration.md:35]: I think somewhere we had some wording about running through the 4-step resource modification cycle for each parameter (where appropriate)

File: group-aws/integration.md [hunk 2]
```diff
@@ -12,167 +12,28 @@ shall be considered a feature request.
 Following the 2.5 release, all new modules MUST have integration
 tests.
 
-Modules marked `preview` SHOULD have integration tests for new
-features.
+Following the 2.10 release, all modules MUST have integration tests.
```
Comment (tremble) [inline group-aws/integration.md:15]: Given the number of modules without integration tests must still feels a little harsh.

File: group-aws/integration.md [hunk 3]
```diff
@@ -12,167 +12,28 @@ shall be considered a feature request.
 Following the 2.5 release, all new modules MUST have integration
 tests.
```
Comment (tremble) [inline group-aws/integration.md:13]:
```suggestion
tests.  Where integration tests are impractical due to the need for non-AWS resources, unit tests
can be used instead.
```

File: group-aws/integration.md (hunk 2 above)
Comment (jillr) [reply group-aws/integration.md:15]: I'd really like to get there, but yeah there's still gaps.  Wanted to try to idea on though.

Comment (jillr) [reply group-aws/integration.md:15]: What do you think of changing the below SHOULD (regarding bug fixes for modules that have existing tests) to a MUST?

File: group-aws/integration.md (hunk 1 above)
Comment (jillr) [reply group-aws/integration.md:35]: What is the bit in https://github.com/ansible-collections/amazon.aws/blob/main/CONTRIBUTING.md?  I'll add that in.

**Comment by jillr** [comment]
@tremble @s-hertel would you both be up for giving this one a review?

**Review by tremble — APPROVED** [review]
These changes look sound to me.

**Comment by jillr** [comment]
I fell down on this one; afaict it's ready to merge we just never pushed that button?  Doing so now, if anyone disagrees well.. file an issue!  :)

**Pull request merged**